pymc4/distributions/continuous.py

```diff
@@ -1174,6 +1177,97 @@ def upper_limit(self):
         return self.conditions["high"]
 
 
+class Flat(ContinuousDistribution):
+    r"""A uniform distribution with support :math:`(-\inf, \inf)`.
+    Used as a uninformative log-likelihood that returns
+    zeros regardless of the passed values.
+    """
+
+    def __init__(self, name, **kwargs):
+        super().__init__(name, **kwargs)
+
+    @staticmethod
+    def _init_distribution(conditions):
+        return tfd.Uniform(low=-np.inf, high=np.inf)
+
+    def log_prob(self, value):
+        # convert the value to tensor
+        value = tf.convert_to_tensor(value)
+        expected = tf.zeros(self.batch_shape + self.event_shape)
+        # check if the event shape matches
+        if len(self.event_shape) and value.shape[-len(self.event_shape) :] != self.event_shape:
+            raise ValueError("values not consistent with the event shape of distribution")
+        # broadcast expected to shape of value
+        if len(value.shape) < len(self.batch_shape + self.event_shape):
+            if (
+                value.shape[: -len(self.event_shape)]
+                != list(reversed(self.batch_shape))[: (len(value.shape) - len(self.event_shape))]
```

Comment:
Why do you need to reverse this?

Reply:
If the ``batch_shape`` is ``(1, 2)``, ``event_shape`` is ``(3, 4)``, and we want to sample ``(2, 3, 4)`` then the target must broadcast to ``(1, 2, 3, 4)``. So, we need to check the values of batch_shape from the rightmost axis.

Reply:
Oh, I see the problem! I have to check the ``batch_shape[len(self.batch_shape) - len(value.shape):]`` instead of reversing it.